src/frame.js

```diff
@@ -0,0 +1,54 @@
+function Frame(multipliers, isFinalFrame) {
+  this.multipliers = multipliers;
+  this.rolls = [];
+  this.score = 0;
+  this.isFinalFrame = isFinalFrame;
+}
+
+Frame.prototype.roll = function(pinsKnocked) {
+  this.rolls.push(pinsKnocked);
+  this._updateScore(pinsKnocked);
+  this._calculateMultipliers();
+};
+
+Frame.prototype.isComplete = function() {
+  if (this.isFinalFrame) {
+    return this._finalFrameIsComplete();
+  } else {
+    return this._notFinalFrameIsComplete();
+  }
+};
+
+Frame.prototype._updateScore = function(pinsKnocked) {
+  let currentMultiplier = this.multipliers.shift();
+  this.score += (pinsKnocked * currentMultiplier);
+};
+
+Frame.prototype._calculateMultipliers = function() {
+  let multipliers = this.multipliers;
```

Comment:
I'm afraid I'm struggling to understand this implementation of multipliers for bonuses. Might be worth discussing.